packages/init/src/index.js

```diff
@@ -130,8 +139,106 @@ const install = async () => {
   });
 };
 
+const listAndSelectTemplate = async () => {
+  
+  const getTemplates = async () => {
+    try {
+
+      // create error response 
+      class HTTPResponseError extends Error {
+        constructor(response, ...args) {
+          super(`HTTP Error Response: ${response.status} ${response.statusText}`, ...args);
+          this.response = response;
+        }
+      }
+      
+      // check response from repo list fetch
+      const checkStatus = response => {
+        if (response.ok) {
+          // response.status >= 200 && response.status < 300
+          return response.json();
+        } else {
+          console.log('Couldn\'t locate any templates, check your connection and try again');
+          throw new HTTPResponseError(response);
+        }
+      };
+
+      const repos = await fetch(PROJECT_API_URL).then(resp => checkStatus(resp));
+
+      // assuming it did resolve but there are no templates listed
+      if (!repos || repos.length === 0) {
+        console.log('Couldn\'t locate any templates, check your connection and try again');
+        return [];
+      }
+
+      const templateRepos = repos.filter(repo => {
+        return repo.name.includes('greenwood-template');
+      });
+      
+      return templateRepos.map(({ clone_url, name }) => {
+        const templateName = name.substring(templateStandardName.length, name.length);
+        return { clone_url, name: templateName };
+      });
+    } catch (err) {
+      throw err;
+    }
+  };
+
+  const templates = await getTemplates();
+
+  // Debug list templates 
+  // console.log('templates', templates);
+
+  const questions = [
+    {
+      type: 'list',
+      name: 'template',
+      message: 'Which template would you like to use?',
+      choices: templates.map(template => template.name),
+      filter(val) {
+        return val.toLowerCase();
+      }
+    }
+  ];
+
+  return inquirer.prompt(questions).then((answers) => {
+    // set the selected template based on the selected template name
+    selectedTemplate = templates.find(template => {
+      return template.name === answers.template;
+    });
+
+    if (selectedTemplate) {
+      console.log('\Installing Selected Template:', selectedTemplate.name);
+    }
+  });
+};
+
+const cloneTemplate = async () => {
+  const git = simpleGit();
+
+  // check if .template directory already exists, if so remove it
+  if (await fs.existsSync(clonedTemplateDir)) {
+    await fs.rmSync(clonedTemplateDir, { recursive: true, force: true });
+  }
+
+  // clone to .template directory 
+  console.log('clone template', selectedTemplate.name, 'to directory', clonedTemplateDir);
+  try {
+    await git.clone(selectedTemplate.clone_url, clonedTemplateDir);
+    templateDir = clonedTemplateDir;
```

Comment:
Should we also clean up `clonedTemplateDir` too after we scaffold the template?  Also, as we've stablished _.greenwood_ as our scratch folder for build output and temp files / operations, would it make sense to keep _.template_ inside of that instead of making extra top level hidden directory in user's project?

Reply:
both good points